src/commands.ts

```diff
@@ -42,12 +44,51 @@ export default function registerCommands(commandObserver: CommandObserver, packa
 					});
 				});
 		}),
+		vscode.commands.registerCommand('pgproj.deploy.current', async (args) => {
+			requireDotNetSdk(dotNetSdkVersion).then(
+				async dotnet => {
+					await deployCurrentProject(args, dotnet, commandObserver);
+				});
+		}),
 		vscode.commands.registerCommand('pgproj.add.new', async (args) => {
-			await addNewPostgreSQLProject(args, packageInfo.maxSupportedPostgreSQLProjectSDK, commandObserver);
+			await addNewPostgreSQLProject(args, packageInfo.maxSupportedPostgreSQLProjectSDK);
 		})
 	];
 }
 
+async function deployCurrentProject(args, dotNetSdk: DotNetInfo, commandObserver:  CommandObserver, cancelToken?: vscode.CancellationToken) {
+	let project = '';
+	if (!args) {
+		const activeEditor = vscode.window.activeTextEditor;
+		if (activeEditor !== undefined) {
+			project = activeEditor.document.uri.fsPath;
```

Comment:
What is this supposed to handle? If you are modifying ```.pgproj``` file right now or it will somehow find parent pgproj when you are editing ```.sql``` and want to deploy?

Reply:
This will get the pgproj file name, if you right click on the editor then we do not get the file name in args and use the editor to get it.

Reply:
I'm still not sure I understand. Yes, it is for the editor, I figured that out. Which editor was my question. If you edit ```.sql``` file it will find the parent ```.pgproj``` or only if you modify ```.pgproj``` in the editor (which is rather uncommon scenario)? I think deploying from ```.sql``` would be nice.

Reply:
only if you right click on .pgproj. As we do not have a project concept here like in Visual Studio where you click on project, I think we should use pgproj that way. Like we only allow Build using pgproj we will allow user to deploy when they right click on it.